tensorboard/webapp/app_routing/store_only_utils.ts

```diff
@@ -34,3 +40,45 @@ export function getCompareExperimentIdAliasSpec(
   }
   return idToDisplayName;
 }
+
+function getRouteParamsFromNavigation(
+  navigation: ProgrammaticalNavigation
+): RouteParams {
+  return navigation.routeKind === RouteKind.COMPARE_EXPERIMENT
+    ? {
+        experimentIds: serializeCompareExperimentParams(
+          navigation.routeParams.aliasAndExperimentIds
+        ),
+      }
+    : navigation.routeParams;
+}
+
+/**
+ * Returns experimentIds from navigation.
+ *
+ * This utility is used by only limited packages. Please refer to visiblity in
+ * BUILD.
+ */
+export function getExperimentIdsFromNavigation(
+  navigation: ProgrammaticalNavigation
+): string[] | null {
+  return getExperimentIdsFromRouteParams(
+    navigation.routeKind,
+    getRouteParamsFromNavigation(navigation)
+  );
+}
+
+/**
+ * Returns routeId from navigation.
+ *
+ * This utility is used by only limited packages. Please refer to visiblity in
+ * BUILD.
+ */
+export function getRouteIdFromNavigation(
+  navigation: ProgrammaticalNavigation
```

Comment:
Instead of `ProgrammaticalNavigation`, you want to use `Route`. Action payload for https://github.com/tensorflow/tensorboard/blob/bc216f499ec934113fdfd0f92a055c448b6daa8e/tensorboard/webapp/app_routing/actions/app_routing_actions.ts#L56-L57.

Reply:
Done, thanks!